src/main/java/goodgenerator/blocks/tileEntity/ComponentAssemblyLine.java

```diff
@@ -0,0 +1,744 @@
+
+    @Override
+    public ITexture[] getTexture(
+            IGregTechTileEntity aBaseMetaTileEntity,
+            byte aSide,
+            byte aFacing,
+            byte aColorIndex,
+            boolean aActive,
+            boolean aRedstone) {
+        if (aSide == aFacing) {
+            if (aActive)
+                return new ITexture[] {
+                    Textures.BlockIcons.getCasingTextureForId(183),
+                    TextureFactory.builder()
+                            .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_ACTIVE)
+                            .extFacing()
+                            .build(),
+                    TextureFactory.builder()
+                            .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_ACTIVE_GLOW)
+                            .extFacing()
+                            .glow()
+                            .build()
+                };
+            return new ITexture[] {
+                Textures.BlockIcons.getCasingTextureForId(183),
+                TextureFactory.builder()
+                        .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY)
+                        .extFacing()
+                        .build(),
+                TextureFactory.builder()
+                        .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_GLOW)
+                        .extFacing()
+                        .glow()
+                        .build()
+            };
+        }
+        return new ITexture[] {Textures.BlockIcons.getCasingTextureForId(183)};
+    }
+
+    @Override
+    public boolean isCorrectMachinePart(ItemStack aStack) {
+        return true;
+    }
+
+    @Override
+    public boolean checkRecipe(ItemStack aStack) {
+        long tVoltage = GT_ExoticEnergyInputHelper.getMaxInputVoltageMulti(getExoticAndNormalEnergyHatchList());
+        long tAmps = GT_ExoticEnergyInputHelper.getMaxInputAmpsMulti(getExoticAndNormalEnergyHatchList());
+        long totalEU = tVoltage * tAmps;
+        ItemStack[] tItems = getCompactedInputs();
+        FluidStack[] tFluids = getCompactedFluids();
+        GT_Recipe foundRecipe = getRecipeMap().findRecipe(getBaseMetaTileEntity(), false, totalEU, tFluids, tItems);
+        if (foundRecipe == null) return false;
+        if (foundRecipe.mSpecialValue > casingTier + 1) return false;
+        if (!foundRecipe.isRecipeInputEqual(true, tFluids, tItems)) return false;
+
+        // Logic for overclocking calculations.
+        double EU_input_tier = Math.log(totalEU) / log4;
+        double EU_recipe_tier = Math.log(foundRecipe.mEUt) / log4;
+        long overclock_count = (long) Math.floor(EU_input_tier - EU_recipe_tier);
+
+        // Vital recipe info. Calculate overclocks here if necessary.
+        EU_per_tick = (long) -(foundRecipe.mEUt * Math.pow(4, overclock_count));
+
+        mMaxProgresstime = (int) (foundRecipe.mDuration / Math.pow(2, overclock_count));
+        mMaxProgresstime = Math.max(1, mMaxProgresstime);
+
+        mOutputItems = foundRecipe.mOutputs.clone();
```

Comment:
`clone()` here is unnecessary

Reply:
fixed in [4d8ec19](https://github.com/GTNewHorizons/GoodGenerator/pull/102/commits/4d8ec19ba2e385e784c500db43f4ae98a089a445)